tests/suit_manifest/Makefile

```diff
@@ -22,15 +21,20 @@ endif
 
 FEATURES_REQUIRED += periph_flashpage
 
-TEST_DATA = $(BIN_DIR)/created
+TEST_DATA = $(MANIFEST_DIR)/created
 BUILDDEPS += $(TEST_DATA)
 
 include $(RIOTBASE)/Makefile.include
 
+CFLAGS += -DBOARD_NAME_UNQ=$(BOARD)
+
 $(call target-export-variables,all,SUIT_TOOL)
 $(call target-export-variables,all,SUIT_SEC)
+$(call target-export-variables,all,MANIFEST_DIR)
+
+.PHONY: $(TEST_DATA)
 
 $(TEST_DATA): $(SUIT_SEC)
-	@mkdir -p $(BIN_DIR)/manifests
+	@mkdir -p $(MANIFEST_DIR)
 	sh create_test_data.sh
 	@touch $@
```

Comment:
I played around with the test when changing keys and found some issue when there were 2 already created keys and switching between them the manifest would not be re-generated, a logic like the following seems to work:

```
CURRENT_TEST_KEY = $(BIN_DIR)/current-test-key

$(CURRENT_TEST_KEY): $(SUIT_SEC) FORCE | $(CLEAN)
	$(Q)cat $< > $@.tmp
	$(Q)rsync -c $@.tmp $@
	$(Q)rm -f $@.tmp

$(TEST_DATA): $(CURRENT_TEST_KEY) | $(CLEAN)
	@mkdir -p $(BIN_DIR)/manifests
	sh create_test_data.sh
	@touch $@
```

The rule checking the key is allays triggered and updates `current-test-key` only if `SUIT_SEC` changed. This is then a requirement of `TEST_DATA` so it will always rebuild the manifests if the `SUIT_SEC` is switched between a couple of keys.

What do you think?

Reply:
Which command sequence do I have to use to reproduce this issue?

Reply:
1.- Compile twice with default key, `create_test_data` should be called once

```
$ make -C tests/suit_manifest/ all
$ make -C tests/suit_manifest/ all
```

2.- Compile twice with different key, `create_test_data` should be called once

```
$ SUIT_KEY=toto make -C tests/suit_manifest/ all
$ SUIT_KEY=toto make -C tests/suit_manifest/ all
```

3.- Both keys now exist, if switching to default key and back to `toto` `create_test_data` should be called both times

```
$ make -C tests/suit_manifest/ all
$ SUIT_KEY=toto make -C tests/suit_manifest/ all
```

Without my suggestion, it wont since both keys already exist `make` will think it doesn't need to re-do  `create_test_data`.

Reply:
Is it possible to either consider this issue unlikely and solve it with a comment in the `README.md` of the test or find a different solution? I strongly prefer to not start copying the secret key into build directories to prevent it from being accidentally leaked.

Reply:
As discussed out-of-band, adding `$(SUIT_PUB_HDR)` to the `$(TEST_DATA)` dependencies fixes this issue.